esp-hal-common/src/rom.rs

```diff
@@ -127,3 +127,21 @@ macro_rules! regi2c_write_mask {
         }
     };
 }
+
+/// Determine the reason that the specified CPU reset
+pub unsafe fn rtc_get_reset_reason(cpu_num: u32) -> u32 {
+    #[cfg(esp32)]
+    const RTC_GET_RESET_REASON: u32 = 0x4000_81d4;
+    #[cfg(esp32c2)]
+    const RTC_GET_RESET_REASON: u32 = 0x4000_0018;
+    #[cfg(esp32c3)]
+    const RTC_GET_RESET_REASON: u32 = 0x4000_0018;
+    #[cfg(esp32s2)]
+    const RTC_GET_RESET_REASON: u32 = 0x4000_ff58;
+    #[cfg(esp32s3)]
+    const RTC_GET_RESET_REASON: u32 = 0x4000_057c;
+
+    let get_reason: fn(u32) -> u32 = core::mem::transmute(RTC_GET_RESET_REASON as usize);
```

Comment:
This comment is orthogonal to the PR, I'm just curious. I believe we already do it this way in the clocks section & esp-wifi, but did we have a reason for transmuting addresses vs using `PROVIDE()` in each chips linker script?

Reply:
I think in the beginning there was just like one function and adding something like `rom_functions.x` for each chip was a bit inconvenient - but the more rom functions we use that might be the nicer approach

Reply:
Yeah I don't think there's any real reason to continue doing this, I was just copy-pasting as usual 😁  In a subsequent PR we can change this.